src/System.Memory/src/System/Buffers/Binary/Reader.cs

```diff
@@ -93,10 +121,11 @@ public static uint ReverseEndianness(uint value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is this really a good idea to mark this work AggresiveInlining? The code for this will be quite big.

Reply:
The code for this is only 6 assembly instructions, as below. Is there a guideline for when we should use aggressive inlining vs. not use aggressive inlining?

```asm
mov   edx,eax  ; eax already contains 'value'
and   edx,0FF00FFh  
and   eax,0FF00FF00h  
rol   edx,18h  
rol   eax,8  
add   edx,eax  ; edx now contains swapped 'value'  
```

Reply:
This is the code for 32-bit. I have specifically commented on 64-bit...

> Is there a guideline for when we should use aggressive inlining vs. not use aggressive inlining?

I would say when the aggressive inlining gives you less than 10% win on microbenchmarks, it is probably not a good idea.

Reply:
I reran the `ReverseEndianness(ulong)` benchmark with no `MethodImplAttribute` at all. In a 64-bit process the throughput was pretty much identical to the inlined version. In a 32-bit process throughput halved. :(

Reply:
@jkotas Would it make sense to put an `#if x86` around the attibute? I don't think we produce arch-specific versions of this assembly so I don't know if the effort of introducing this split into the build process would be worthwhile.

Reply:
> In a 32-bit process throughput halved

That's sounds weird. Do you have explanation for it?

> I don't think we produce arch-specific versions

Right. We do not build arch-specific assemblies in CoreFX in general.

Reply:
`if (IntPtr.Size == 8)` is `[Intrinsic]` and gives same effect?

Reply:
@jkotas Looks like because on x64 it's a simple fast call instruction with minimal prolog / epilog, but on x86 trying to pass a 64-bit value across the method call boundary results in the parameter being spilled onto the stack. Disassembly shows there's a lot of indirection while trying to reconstitute the long on the other side. I don't know why codegen doesn't just split the value across two registers and avoid the whole indirection mess.

Reply:
Sounds like a codegen bug that should be fixed.

Reply:
@jkotas How would you like me to proceed with regards to this PR? I can remove the attribute, but it'll regress performance significantly from the existing code (which does have the attribute). I can check `IntPtr.Size` and perform something clever based on the result, but this would complicate the code. I don't think this particular code path will be used by a very large number of callers - just serializers, really, and even then only if that a ulong field appears anywhere. So the number of times the inlined codegen ends up being duplicated should hopefully be very small in practice.

Reply:
I would keep the AggresiveInling and open codegen bug in CoreCLR repo on the inefficient code that you have observed.

Reply:
@GrabYourPitchforks, can you please link the issue here once it has been created?